dfget/core/uploader/uploader_util.go

```diff
@@ -104,14 +116,19 @@ func transFile(f *os.File, w http.ResponseWriter, start, readLen int64) error {
 			return nil
 		}
 
-		if int64(num) > remain {
-			w.Write(buf[:remain])
-		} else {
-			w.Write(buf[:num])
+		length := int64(num)
```

Comment:
* remove line `// TODO: limit the read rate.`
* these codes don't support to upload empty file:
```go
		if num == 0 {
			if total == 0 {
				return fmt.Errorf("content is empty")
			}
			return nil
		}
```